src/annotator/plugin/bucket-bar-js.js

```diff
@@ -0,0 +1,52 @@
+import { getBoundingClientRect } from '../highlighter';
+
+// FIXME: Temporary duplication of size constants between here and BucketBar
+const BUCKET_SIZE = 16; // Regular bucket size
+const BUCKET_NAV_SIZE = BUCKET_SIZE + 6; // Bucket plus arrow (up/down)
+const BUCKET_TOP_THRESHOLD = 115 + BUCKET_NAV_SIZE; // Toolbar
+
+/**
+ * Find the closest valid anchor in `anchors` that is offscreen in the direction
+ * indicated.
+ *
+ * @param {Object[]} anchors
+ * @param {'up'|'down'} direction
+ * @return {Object|null} - The closest anchor or `null` if no valid anchor found
+ */
+export function findClosestOffscreenAnchor(anchors, direction) {
+  const { next: closest } = anchors.reduce(
+    (acc, anchor) => {
+      if (!anchor.highlights?.length) {
+        return acc;
+      }
+
+      const top = getBoundingClientRect(anchor.highlights).top;
+
+      if (direction === 'up' && top >= BUCKET_TOP_THRESHOLD) {
+        // We're headed up but the anchor is already below the
+        // visible top of the bucket bar: it's not our guy
+        return acc;
+      } else if (
+        direction === 'down' &&
+        top <= window.innerHeight - BUCKET_NAV_SIZE
+      ) {
+        // We're headed down but this anchor is already above
+        // the usable bottom of the screen: it's not our guy
+        return acc;
+      }
+
+      const dir = direction === 'up' ? +1 : -1;
+      if (acc.next === null || acc.top * dir < top * dir) {
```

Comment:
I see this was taken from the existing code, but I found the right-hand side of this expression tricky to think about quickly. IMO it would be better to write out the two cases more directly:

```
(direction === 'up' && top > acc.top) || (direction === 'down' && top < acc.top)
```

Reply:
Thumbs way way way way up on this suggestion. I...almost did something like this in my first pass but (inexplicably?) felt like maybe I was taking too much liberty?